--- SafeInt-3.0.28a.orig/Test/CompileTest.cpp	2026-02-08 17:43:29
+++ SafeInt-3.0.28a/Test/CompileTest.cpp	2026-02-08 17:44:28
@@ -11,6 +11,26 @@
 #endif
 
 #include "../SafeInt.hpp"
+#if CPLUSPLUS_STD == CPLUSPLUS_17
+#include <variant>
+#include <string>
+
+template<typename... T>
+class Union {
+public:
+	Union() = default;
+	Union& operator=(Union&&) noexcept = default;
+
+	template<typename F>
+	Union& operator=(F&& t) {
+		value = std::forward<F>(t);
+		return *this;
+	}
+
+	std::variant<T...> value;
+};
+#endif
+
 
 
 void ConstExpr();
@@ -260,24 +280,6 @@
 void MoveRegression()
 {
 	#if CPLUSPLUS_STD == CPLUSPLUS_17
-	#include <variant>
-	#include <string>
-
-	template<typename... T>
-	class Union {
-	public:
-		Union() = default;
-		Union& operator=(Union&&) noexcept = default;
-
-		template<typename F>
-		Union& operator=(F&& t) {
-			value = std::forward<F>(t);
-			return *this;
-		}
-
-		std::variant<T...> value;
-	};
-
 	Union<std::string, SafeInt<int64_t>> x;
 	Union<std::string, SafeInt<int64_t>> y;
 	x = std::move(y);
