diff --git a/configure.ac b/configure.ac
index aecbea6..ceb0239 100644
--- a/configure.ac
+++ b/configure.ac
@@ -445,6 +445,14 @@ if test "x$with_systemdsystemunitdir" != xno; then
         AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])
 fi
 
+#Setup the sysusersdir for systemd stuff
+AC_ARG_WITH([systemd-sysusersdir],
+        AS_HELP_STRING([--with-systemd-sysusersdir=DIR], [Directory for systemd sysusers files]),
+        [], [with_systemdsysusersdir=$($PKG_CONFIG --variable=sysusersdir systemd)])
+if test "x$with_systemdsysusersdir" != xno; then
+        AC_SUBST([systemdsysusersdir], [$with_systemdsysusersdir])
+fi
+
 #Setup the tmpfilesdir for systemd stuff
 AC_ARG_WITH([systemd-tmpfilesdir],
         AS_HELP_STRING([--with-systemd-tmpfilesdir=DIR], [Directory for systemd temporary files]),
@@ -502,6 +510,7 @@ AC_CONFIG_FILES([
     tools/bash_completion/Makefile
     packaging/Makefile
     packaging/daemon/Makefile
+    packaging/daemon/user_group/Makefile
     packaging/libstoragemgmt.spec
     doc/man/Makefile
     test/Makefile])
diff --git a/packaging/daemon/Makefile.am b/packaging/daemon/Makefile.am
index 7959c95..328e4b0 100644
--- a/packaging/daemon/Makefile.am
+++ b/packaging/daemon/Makefile.am
@@ -7,3 +7,5 @@ systemdtmpfiles_DATA = libstoragemgmt.conf
 endif
 
 EXTRA_DIST = libstoragemgmt.service libstoragemgmt.conf libstoragemgmtd
+
+SUBDIRS = user_group
diff --git a/packaging/daemon/user_group/Makefile.am b/packaging/daemon/user_group/Makefile.am
new file mode 100644
index 0000000..9c8fc9c
--- /dev/null
+++ b/packaging/daemon/user_group/Makefile.am
@@ -0,0 +1,9 @@
+
+if HAVE_SYSTEMD
+
+systemdsysusers_DATA = libstoragemgmt.conf
+
+endif
+
+EXTRA_DIST = libstoragemgmt.conf
+
diff --git a/packaging/daemon/user_group/libstoragemgmt.conf b/packaging/daemon/user_group/libstoragemgmt.conf
new file mode 100644
index 0000000..9aa2df5
--- /dev/null
+++ b/packaging/daemon/user_group/libstoragemgmt.conf
@@ -0,0 +1,2 @@
+#Type Name            ID               GECOS                                HomeDir Shell
+u!    libstoragemgmt  -                "daemon account for libstoragemgmt"
